scipy/stats/_continuous_distns.py

```diff
@@ -7943,27 +7944,119 @@ class argus_gen(rv_continuous):
+        # expected number of iterations is c with
+        # c = -np.expm1(-0.5 * chi**2) * chi / (_norm_pdf_C * _argus_phi(chi))
+        # note that G can be inverted easily, so we can sample
+        # rvs from this distribution
+        # G_inv(y) = sqrt(2 * log(1 + (exp(chi**2 / 2) - 1) * y) / chi**2)
+        # to avoid an overflow of exp(chi**2 / 2), it is convenient to write
+        # G_inv(y) = sqrt(1 + 2 * log(exp(-chi**2 / 2) * (1-y) + y) / chi**2)
+        #
+        # if chi > 2.611:
+        # use ratio of uniforms method applied to a transformed variable of X
+        # (X is ARGUS with parameter chi):
+        # Y = chi * sqrt(1 - X**2) has density proportional to
+        # u**2 * exp(-u**2 / 2) on [0, chi] (Maxwell distribution conditioned
+        # on [0, chi]). Apply ratio of uniforms to this density to generate
+        # samples of Y and convert back to X
+        # the expected number of iterations is below 2.2 for all chi > 0.
+        # the value 2.611 can be found by comparing the rejection constants
+        # of the ratio of uniforms method and the rejection method.
+        # it can be shown that one should use the rejection method for chi
+        # less than 2.611 and the ratio of uniforms method for larger chi.
+        # the value can be found by comparing the rejection constants
+        # of the ratio of uniforms method and the rejection method which gives
+        # chi*(1 - exp(-0.5*chi**2)) = 8*sqrt(2)*exp(-1.5) => chi ~ 2.611
+
+        if chi <= 2.611:
+            # use rejection method
+            size1d = tuple(np.atleast_1d(numsamples))
+            N = int(np.prod(size1d))
+            x = np.zeros(N)
+            echi = np.exp(-chi**2 / 2)
+            simulated = 0
+            while simulated < N:
+                k = N - simulated
+                u = self._random_state.uniform(size=k)
+                v = self._random_state.uniform(size=k)
+                # acceptance condition: u <= h(G_inv(v)). This simplifies to
+                z = 2 * np.log(echi * (1 - v) + v) / chi**2
+                accept = (u**2 + z <= 0)
+                num_accept = np.sum(accept)
+                if num_accept > 0:
+                    # rvs follow a distribution with density g: rvs = G_inv(v)
+                    rvs = np.sqrt(1 + z[accept])
+                    x[simulated:(simulated + num_accept)] = rvs
+                    simulated += num_accept
+
+            if self._size == ():
+                # return scalar if rvs() is called without size specified
+                return x[0]
+            return np.reshape(x, size1d)
+        else:
+            # use ratio of uniforms method
+            def f(x):
+                return np.where((x >= 0) & (x <= chi),
+                                np.exp(2*np.log(x) - x**2/2), 0)
+
+            umax = np.sqrt(2) / np.exp(0.5)
+            vmax = 4 / np.exp(1)
+            z = rvs_ratio_uniforms(f, umax, 0, vmax, size=numsamples)
```

Comment:
This call to does `rvs_ratio_uniforms` not pass in `self._random_state` and needs to.

Reply:
good catch, fixed